src/dropdown/dropdown.ts

```diff
@@ -12,11 +17,13 @@ import {NgbDropdownConfig} from './dropdown-config';
     '[class.dropup]': 'up',
     '[class.show]': 'isOpen()',
     '(keyup.esc)': 'closeFromOutsideEsc()',
-    '(document:click)': 'closeFromOutsideClick($event)'
+    '(document:click)': 'closeFromOutsideClick($event)',
+    '(keydown)': 'keyboardEvent($event)'
```

Comment:
I think one thing that may help is binding to the particular keys here, i.e.
```ts
'(keydown.ArrowUp)': 'moveUp()',
'(keydown.ArrowDown)': 'moveDown()'
```
Under the hood, I believe Angular creates only one event listener and does the appropriate things. This avoids the need to pass in the event object and have a bunch of conditional logic in your method.